tests/python_package_test/test_engine.py

```diff
@@ -1545,6 +1545,59 @@ def test_refit():
     assert err_pred > new_err_pred
 
 
+def test_refit_dataset_params():
+    # check refit accepts dataset_params
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    lgb_train = lgb.Dataset(X_train, y_train)
+    train_params = {
+        'objective': 'binary',
+        'metric': 'binary_logloss',
+        'verbose': -1,
+        'min_data': 10
+    }
+    gbm = lgb.train(train_params, lgb_train, num_boost_round=20)
+    non_weight_err_pred = log_loss(y_test, gbm.predict(X_test))
+    dataset_params = {
+        'linear_tree': False,
+        'max_bin': 255,
+        'max_bin_by_feature': None,
+        'min_data_in_bin': 3,
+        'bin_construct_sample_cnt': 200000,
+        'data_random_seed': 1,
+        'is_enable_sparse': True,
+        'enable_bundle': True,
+        'use_missing': True,
+        'zero_as_missing': False,
+        'feature_pre_filter': True,
+        'pre_partition': False,
+        'two_round': False,
+        'header': False,
+        'label_column': "",
+        'weight_column': "",
+        'group_column': "",
+        'ignore_column': "",
+        'forcedbins_filename': "",
+        'save_binary': False,
+        'precise_float_parser': False,
+        'parser_config_file': "",
+    }
```

Comment:
Please see my comment from https://github.com/microsoft/LightGBM/pull/4894#discussion_r773321120.

If these parameter values are not all necessary for this test, please remove them from this code. If they are necessary, could you please explain why?

Reply:
I'm setting them as default to test that `Dataset` "accepts" them, think of just leaving these three as they influence the behavior
```
'max_bin': 255,
'min_data_in_bin': 3,
'data_random_seed': 1
```